### PR TITLE
Adds 2 delete buttons, one on the list and one on the actual reminder(Closes Issue #18)

### PR DESCRIPTION
**Pull request by MilkMan90**
@martensonbj @Tman22 @nfosterky @stevekinney

## Purpose

User can remove/delete a reminder.

Users should be able to remove a reminder from both the list of reminders or from an individual reminder’s page.

If a reminder is removed from the individual reminder’s page, then the application should reroute them back to the '/reminders' route.

## Approach

We added a controller for the reminder and reminder list which deletes the record for that specific reminder on click.

### Learning

[Creating Upating and Deleting Records (ember docs)](https://guides.emberjs.com/v2.0.0/models/creating-updating-and-deleting-records/)

We also sat down with Taylor to work through some refactoring of our model hooks.

### Test coverage 

Wrote both tests to make sure that the item is removed from the page after clicking on each new delete button

### Follow-up tasks

Isue #21

### CODE REVIEW

File: app/components/reminder-form.js [hunk 1]
```diff
@@ -3,25 +3,10 @@ import Ember from 'ember';
...
-          reminderModel.save()
-          return this.rerouteBack(reminderModel)
-        });
-      } else {
-          this.get('store').createRecord('reminder', reminder).save().then(() => {
-            this.setProperties({ title: '', date: '', notes: '' });
-          });
-        }
+      reminderModel.save()
+      this.sendAction("action", this.get('model'));
```
Comment (martensonbj) [inline app/components/reminder-form.js]: Nice refactor! Could us a bit of re-alignment in your indentation

File: app/templates/components/reminder-form.hbs [hunk 2]
```diff
@@ -1,21 +1,21 @@
...
+   {{input class="input-notes" type="textarea" value=model.notes}}
  </label>
 
   {{input class="input-submit" type="submit" value="save"}}
 
 </form>
 
 {{#if edit}}
-  <button class="input-revert {{if reminder.hasDirtyAttributes "enabled"}}" {{action 'rollback' reminder on="click"}}>Revert</button>
+  <button class="input-revert {{if reminder.hasDirtyAttributes "enabled"}}" {{action 'rollback' model on="click"}}>Revert</button>
```
Comment (martensonbj) [inline app/templates/components/reminder-form.hbs]: `onClick` is the default function for a button - you can remove the `on="click"` bit here

File: app/templates/reminders.hbs [hunk 3]
```diff
@@ -2,12 +2,13 @@
...
   {{#if model}}
     <ul class="reminder-list">
-      {{#each model as |reminder|}}
+      {{#each savedReminders as |reminder|}}
         {{#link-to 'reminders.reminder' reminder}}
           <li class='spec-reminder-item'>
               {{reminder.title}}
           </li>
         {{/link-to}}
+        <button class="reminders-delete" {{action 'deleteReminder' reminder on="click"}}>X</button>
```
Comment (martensonbj) [inline app/templates/reminders.hbs]: See above - dont need the `on="click"`

File: app/templates/reminders/reminder.hbs [hunk 4]
```diff
@@ -7,6 +7,7 @@
   {{#link-to 'reminders.edit' reminder}}
     <button class="reminder-edit">Edit</button>
   {{/link-to}}
+  <button class="reminder-delete" {{action 'deleteReminder' reminder on="click"}} >Delete</button>
```
Comment (martensonbj) [inline app/templates/reminders/reminder.hbs]: Remove `on="click"`

File: tests/unit/controllers/reminders-test.js [hunk 5]
```diff
@@ -0,0 +1,12 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/controllers/reminders-test.js]: Remove unused test files

File: tests/unit/controllers/reminders/reminder-test.js [hunk 6]
```diff
@@ -0,0 +1,12 @@
+import { moduleFor, test } from 'ember-qunit';
```
Comment (martensonbj) [inline tests/unit/controllers/reminders/reminder-test.js]: See above

**Comment by MilkMan90** [comment]
@martensonbj fixed =)

**Pull request merged**